### PR TITLE
UI improvement

### PR DESCRIPTION
**Pull request by virginieky**
Components fix and alignment to improve the UI

### CODE REVIEW

File: docs/stories/components/Checkbox.stories.js [hunk 1]
```diff
@@ -62,10 +62,7 @@ function CheckboxStory() {
         <ul>
           {Object.keys(checkboxes).map(current => {
             return (
-              <li
-                key={current}
-                style={{ marginBottom: 1, paddingTop: 18, paddingLeft: 15 }}
-              >
+              <li key={current} style={{ paddingTop: 18, paddingLeft: 15 }}>
```
Comment (HichamELBSI) [inline docs/stories/components/Checkbox.stories.js]: We should avoid these kinds of object literal to improve the performance of the application. This is similar to the anonymous functions. Object literals don't have a persistent memory space, so the component will need to allocate a new location in memory for this object whenever the component re-renders.

Comment (HichamELBSI) [reply docs/stories/components/Checkbox.stories.js]: You don't need to fix them for the moment but we need to keep in mind those best practices.

File: docs/stories/components/Button.stories.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ import { Button } from '@buffetjs/core';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import {
   faPlus,
-  faTrash,
+  faTrashAlt,
```
Comment (soupette) [inline docs/stories/components/Button.stories.js:47]: You don't need this import in the example as you don't use it in the code. Same for the `faBaseBall`

File: docs/stories/components/Checkbox.stories.js (hunk 1 above)
Comment (soupette) [reply docs/stories/components/Checkbox.stories.js:65]: I agree plus it seems that the baseline is not kept here

File: docs/stories/components/DatePicker.stories.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ import moment from 'moment';
 import { DatePicker } from '@buffetjs/core';
 
 function Example() {
-  const [value, setValue] = useState(moment());
+  const [value, setState] = useState(moment([2020, 0, 11]));
```
Comment (soupette) [inline docs/stories/components/DatePicker.stories.js:31]: If you change the name here you also need to change in the callback of the example.

File: packages/buffetjs-core/src/components/DatePicker/index.js [hunk 4]
```diff
@@ -35,134 +49,131 @@ class DatePicker extends React.PureComponent {
...
+    dispatch({
+      type: 'SET_DATE',
+      date,
+    });
+
+    dispatch({
+      type: 'SET_DISPLAYED_DATE',
+      displayedDate,
+    });
+  }, []);
```
Comment (soupette) [inline packages/buffetjs-core/src/components/DatePicker/index.js:61]: You are missing some dependencies. You can either disable the eslint-rule here to let the reviewer know it is intended or add them to the dependencies array.

File: packages/buffetjs-core/src/components/DatePicker/index.js [hunk 5]
```diff
@@ -35,134 +49,131 @@ class DatePicker extends React.PureComponent {
...
+    handleDateChange(dateValue);
+
+    toggleDatepicker(false);
   };
 
-  handleOutsideClick = () => {
-    this.setState({
-      visible: false,
-    });
+  const handleOutsideClick = () => {
```
Comment (soupette) [inline packages/buffetjs-core/src/components/DatePicker/index.js:116]: I think this one is not really needed as you can directly use `toggleDatePicker(false)`

File: packages/buffetjs-styles/src/components/Button/index.js [hunk 6]
```diff
@@ -62,18 +63,24 @@ const Button = styled.button`
...
-
   ${({ color }) =>
     color === 'success' &&
     `
-      min-width: ${sizes.button.minWidth.small};
+      min-width: ${sizes.button.minWidth};
     `}
+  
+    // FontAwesome icons
+    > span svg {
```
Comment (soupette) [inline packages/buffetjs-styles/src/components/Button/index.js:73]: The svg selector is overridden by the second one so it not needed.

File: packages/buffetjs-styles/src/components/Button/index.js [hunk 7]
```diff
@@ -62,18 +63,24 @@ const Button = styled.button`
...
-    `}
-
   ${({ color }) =>
     color === 'success' &&
     `
-      min-width: ${sizes.button.minWidth.small};
+      min-width: ${sizes.button.minWidth};
     `}
+  
+    // FontAwesome icons
```
Comment (soupette) [inline packages/buffetjs-styles/src/components/Button/index.js:72]: Can you comment it as follows: `/* my comment */` so we won't have to modify this file when we add stylelint?

File: packages/buffetjs-styles/src/components/Paging/index.js [hunk 8]
```diff
@@ -31,7 +31,7 @@ const Paging = styled.div`
       }
       &.selected {
         .page-link {
-          font-weight: 700;
+          font-weight: ${sizes.fontWeight.black};
         }
       }
       // Left & right arrows
```
Comment (soupette) [inline packages/buffetjs-styles/src/components/Paging/index.js:37]: Same here about the comment format.

File: packages/buffetjs-styles/src/components/TimeList/index.js [hunk 9]
```diff
@@ -60,10 +60,10 @@ const TimeList = styled.ul`
       z-index: 0;
       width: 100%;
       height: 100%;
-      padding: ${sizes.margin * 0.3}px 0 ${sizes.margin * 0.3}px
+      padding: ${sizes.margin * 0.4}px 0 ${sizes.margin * 0.2}px
```
Comment (soupette) [inline packages/buffetjs-styles/src/components/TimeList/index.js:63]: Why do we have a mix between `rem` and `px` here? Also it might be great if the `sizes` file includes the units completely or does not at all. Maybe we should make it uniform in another PR.

File: docs/stories/components/Button.stories.js (hunk 2 above)
Comment (virginieky) [reply docs/stories/components/Button.stories.js:47]: We use all of them in the code below

Comment (soupette) [reply docs/stories/components/Button.stories.js:47]: If I am not mistaken these imports concern the Example which only uses the `faTrash` the imports for the story are set in the top of the file.

Comment (soupette) [reply docs/stories/components/Button.stories.js:47]: If you want to leave them you can add more examples though.

Comment (virginieky) [reply docs/stories/components/Button.stories.js:47]: Ok just understood

File: packages/buffetjs-styles/src/components/Button/index.js (hunk 6 above)
Comment (virginieky) [reply packages/buffetjs-styles/src/components/Button/index.js:73]: It is needed because this one is about FontAwesome svgs, which are in a span, and the second one is for our custom svgs. The behavior is not the same for each and the second one isn't overridden because the selector is different.

File: packages/buffetjs-styles/src/components/TimeList/index.js (hunk 9 above)
Comment (virginieky) [reply packages/buffetjs-styles/src/components/TimeList/index.js:63]: For now, the sizes values are made for px. But I agree, we should uniform all the units and sizes soon in another PR!

File: packages/buffetjs-styles/src/components/Button/index.js (hunk 6 above)
Comment (soupette) [reply packages/buffetjs-styles/src/components/Button/index.js:73]: So which `vertical-align` property is needed for the `svg` ?

File: packages/buffetjs-styles/src/components/Button/index.js [hunk 10]
```diff
@@ -65,8 +66,21 @@ const Button = styled.button`
...
-      min-width: ${sizes.button.minWidth.small};
+      min-width: ${sizes.button.minWidth};
     `}
+  
+    /* FontAwesome icons */
+    > span svg {
+      font-size: 10px;
+      vertical-align: initial;
+    }
+    /* Custom icons */
```
Comment (soupette) [inline packages/buffetjs-styles/src/components/Button/index.js:77]: Can you apply the comments formatting for the entire file?

File: packages/buffetjs-styles/src/components/Button/index.js (hunk 6 above)
Comment (virginieky) [reply packages/buffetjs-styles/src/components/Button/index.js:73]: As I said, for a direct children selected by `> span` and which is a Custom icons, it baseline. For a children of a span which is a FontAwesome icon (it's the way they are), it's initial.

**Comment by soupette** [comment]
> Here are some noticeable regressions:
> 
> * The baseline is not kept in most of the components from the Storybook
> * The DateTime component does not work anymore as we cannot select a time, can you check?
> 
> Regarding the DatePicker if you update your branch with the develop one it might fix the issue, can you check?

I have tested the DateTime component it seemed that your branch was not up-to-date with develop. Everything is ok now.
